src/components/LDRouteGuard.vue

```diff
@@ -8,14 +8,20 @@ import ldRedirectMixin from '../mixins/ldRedirect';
 export default {
   mixins: [ldRedirectMixin()],
   props: {
-    component: { type: Object, required: true },
+    component: { type: [Object, Promise], required: true },
     requiredFeatureFlag: { type: String, required: true },
     to: { type: [String, Object], required: true },
   },
   computed: {
     show() {
       return this.$ld.ready && this.$ld.flags[this.requiredFeatureFlag];
     },
+    componentIsPromise() {
+      return !!this.component && typeof this.component.then === 'function';
+    },
```

Comment:
Sorry, I wasn't clear. Let me try again...

Why expose this as a property instead of a local variable in the one place it's used?

Reply:
No reason in particular. I'll change ~